comp/core/secrets/secretsimpl/secrets.go

```diff
@@ -299,6 +327,59 @@ func (r *secretResolver) resolve(data []byte, origin string, notifyCb secrets.Re
 	return finalConfig, nil
 }
 
+// whitelistHandles restricts what config settings may be updated
+// tests can override this to exercise functionality
+var whitelistHandles = []string{"api_key"}
```

Comment:
I prefer it stay close to where it's being used in this case, for a few reasons:
1) This data is only used for refreshing secrets, not for the larger secrets functionality. If it were at the top that may confuse a reader into thinking it's more important / more generally applicable to the secrets feature.
2) This is only used in a single place in the file, and needs to be top-level only so that tests can override it. If it were used multiple places in the file it would make more sense at the top.
3) This is theoretically only needed temporarily, for the initial phase of the api_key refresh feature. Long-term it may not be needed, once arbitrary parts of configuration can be refreshed.

After thinking about this more, I also moved `secretInfoTmpl` to be closer to the single place where it is used. This used to be closer but the component conversion moved it away.

The point about `authorizedConfigPathsCore` is a good one. Added a comment linking it to that variable to hopefully prevent bugs in the future. I don't think these two values should be merged just yet. Each is meant for specific features, that may be closely aligned, but aren't necessarily the same thing.

Reply:
Fair points about the position of the variable :+1:

Reply:
> I don't think these two values should be merged just yet. Each is meant for specific features, that may be closely aligned, but aren't necessarily the same thing.

Yes I agree, we can consider later if they should always be the same set of paths or not